test/suites/integration/deregister_targets_test.go

```diff
@@ -42,33 +42,33 @@ var _ = Describe("Deregister Targets", func() {
 			service,
 			deployment,
 		)
-		time.Sleep(3 * time.Minute) // Wait for creation of VPCLattice resources
-
-		// Verify VPC Lattice Service exists
-		vpcLatticeService = testFramework.GetVpcLatticeService(ctx, pathMatchHttpRoute)
-
-		Expect(*vpcLatticeService.DnsEntry).To(ContainSubstring(
-			latticestore.LatticeServiceName(pathMatchHttpRoute.Name, pathMatchHttpRoute.Namespace)))
+		Eventually(func(g Gomega) {
+			// Put vpcLatticeService verification logic in the Eventually block(), because the controller need some time to create vpcLattice resource
+			vpcLatticeService = testFramework.GetVpcLatticeService(ctx, pathMatchHttpRoute)
```

Comment:
hidden assertions in GetVpcLatticeService, please use latticeClient directly (same for other comments about nested assertions)